### PR TITLE
Verify refactor

### PR DESCRIPTION
**Pull request by judy2k**
Moved the contents of the verify package under `com.nexmo.client`.

Refactored the `verify`, `check`, and `search` endpoints to use the new HttpWrapper & Auth framework.

## Contibution Checklist
* [x] Unit tests!
* [ ] Updated [CHANGELOG.md]
* [x] My name is in [CONTRIBUTORS.md]

### CODE REVIEW

**Comment by judy2k** [comment]
You can ignore the codecov/changes failure - it's because I removed a couple of now-redundant tests, and it's upset.

It's a huge PR, but a lot of the changes were automated. Areas to consider most in review IMO are:

* Client classes
* Endpoint & Method classes - this is where the majority of refactoring took place

It's worth noting that for a couple of these Endpoints, where there was only one Method (they only support POST), I made the Endpoint subclass AbstractMethod, and thus they fill both roles. I'm not sure I like this shortcut - let me know what you think.

File: src/main/java/com/nexmo/client/verify/SearchRequest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.nexmo.client.verify;
+
+public class SearchRequest {
+    /**
+     * Number of maximum request IDs that can be searched for.
+     */
+    private static final int MAX_SEARCH_REQUESTS = 10;
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/SearchRequest.java]: I just realized that you just moved this file so you didn't write this. Still something to think about.

File: src/main/java/com/nexmo/client/verify/endpoints/CheckEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import com.nexmo.client.NexmoClientException;
+import com.nexmo.client.verify.CheckResult;
+import com.nexmo.client.verify.CheckRequest;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+
+public class CheckEndpoint {
+    private static final Log log = LogFactory.getLog(CheckEndpoint.class);
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/endpoints/CheckEndpoint.java]: It doesn't look like we're using this. Should it be removed?

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
+package com.nexmo.client.verify.endpoints;/*
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java]: Fix this copyright comment

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            String name = node.getNodeName();
+            if ("event_id".equals(name)) {
+                eventId = XmlUtil.stringValue(node);
+            } else if ("status".equals(name)) {
+                String str = XmlUtil.stringValue(node);
+                try {
+                    if (str != null)
+                        status = Integer.parseInt(str);
+                } catch (NumberFormatException e) {
+                    log.error("xml parser .. invalid value in <status> node [ " + str + " ] ");
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java]: Could this error log be worded better?

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                    status = BaseResult.STATUS_INTERNAL_ERROR;
+                }
+            } else if ("price".equals(name)) {
+                String str = XmlUtil.stringValue(node);
+                try {
+                    if (str != null)
+                        price = Float.parseFloat(str);
+                } catch (NumberFormatException e) {
+                    log.error("xml parser .. invalid value in <price> node [ " + str + " ] ");
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java]: Same here

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    @Override
+    protected Class[] getAcceptableAuthMethods() {
+        return ALLOWED_AUTH_METHODS;
+    }
+
+    @Override
+    public RequestBuilder makeRequest(VerifyRequest request) throws NexmoClientException, UnsupportedEncodingException {
+        RequestBuilder result = RequestBuilder.post(this.uri)
+                .addParameter("number", request.getNumber())
+                .addParameter("brand", request.getBrand());
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyEndpoint.java]: Is `number` or `brand` required? If so, should we throw an error if they're null?

File: src/main/java/com/nexmo/messaging/sdk/messages/Message.java [hunk 7]
```diff
@@ -227,7 +227,7 @@ public int getWapPushValidity() {
     }
 
     /**
-     * @return com.nexmo.messaging.sdk.messages.parameters.MessageClass The message class that is to be applied to this message.
+     * @return com.nexmo.messaging.verify.messages.parameters.MessageClass The message class that is to be applied to this message.
```
Comment (ChrisGuzman) [inline src/main/java/com/nexmo/messaging/sdk/messages/Message.java]: Just curious did you do this or did the IDE? I don't know how smart intellij is

File: src/test/java/com/nexmo/client/verify/VerifyClientTest.java [hunk 8]
```diff
@@ -104,22 +89,26 @@ public void testVerify() throws Exception {
...
-                "447700900999",
-                "TestBrand",
-                "15555215554", 6, Locale.US);
-        assertEquals(VerifyResult.STATUS_COMMS_FAILURE, r.getStatus());
+        try {
+            VerifyResult r = client.verify(
+                    "447700900999",
+                    "TestBrand",
+                    "15555215554", 6, Locale.US);
+            fail();
```
Comment (ChrisGuzman) [inline src/test/java/com/nexmo/client/verify/VerifyClientTest.java]: Can you leave a message here about why the test should fail if the error is not caught?

**Comment by ChrisGuzman** [comment]
I'm cool with making some of the `Endpoint` classes subclassing `AbstractMethod`. Maybe in a future under the hood release we should rename `AbstractMethod`.

**Comment by ChrisGuzman** [comment]
Also, you should update CHANGELOG.md

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java (hunk 4 above)
Comment (judy2k) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java]: We've got a separate task for logging - I'm going to leave these for the moment.

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java (hunk 5 above)
Comment (judy2k) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyCheckMethod.java]: As above ;-)

File: src/main/java/com/nexmo/client/verify/endpoints/VerifyEndpoint.java (hunk 6 above)
Comment (judy2k) [inline src/main/java/com/nexmo/client/verify/endpoints/VerifyEndpoint.java]: The constructor to `VerifyRequest` ensure's these are non-null, so an exception is thrown before this point.

File: src/main/java/com/nexmo/messaging/sdk/messages/Message.java (hunk 7 above)
Comment (judy2k) [inline src/main/java/com/nexmo/messaging/sdk/messages/Message.java]: Yeah, IntelliJ did this. :heart: IntelliJ

File: src/main/java/com/nexmo/client/verify/endpoints/CheckEndpoint.java (hunk 2 above)
Comment (judy2k) [inline src/main/java/com/nexmo/client/verify/endpoints/CheckEndpoint.java]: I _tend_ to leave them in, but that's mainly planning for future laziness, so I'll remove it.

**Pull request merged**